src/network/hobbits/network.ts

```diff
@@ -0,0 +1,159 @@
+      this.rpc = new HobbitsConnectHandler(this.bootnodes, logger);
+      resolve();
+    });
+  }
+
+
+  // pubsub
+  public subscribeToBlocks(): void {
+    // this.pubsub.subscribe(BLOCK_TOPIC);
+  }
+  public subscribeToAttestations(): void {
+    // this.pubsub.subscribe(ATTESTATION_TOPIC);
+  }
+  public subscribeToShardAttestations(shard: Shard): void {
+    // this.pubsub.subscribe(shardSubnetAttestationTopic(shard));
+  }
+  public unsubscribeToBlocks(): void {
+    // this.pubsub.unsubscribe(BLOCK_TOPIC);
+  }
+  public unsubscribeToAttestations(): void {
+    // this.pubsub.unsubscribe(ATTESTATION_TOPIC);
+  }
+  public unsubscribeToShardAttestations(shard: Shard): void {
+    // this.pubsub.unsubscribe(shardSubnetAttestationTopic(shard));
+  }
+  public async publishBlock(block: BeaconBlock): Promise<void> {
+    // await promisify(this.pubsub.publish.bind(this.pubsub))(
+    //     BLOCK_TOPIC, serialize(block, BeaconBlock));
+  }
+  public async publishAttestation(attestation: Attestation): Promise<void> {
+    // await promisify(this.pubsub.publish.bind(this.pubsub))(
+    //     ATTESTATION_TOPIC, serialize(attestation, Attestation));
+  }
+  public async publishShardAttestation(attestation: Attestation): Promise<void> {
+    // await promisify(this.pubsub.publish.bind(this.pubsub))(
+    //     shardSubnetAttestationTopic(attestation.data.shard), serialize(attestation, Attestation));
+  }
+  private handleIncomingBlock(msg: any): void {
+    try {
+      const block: BeaconBlock = deserialize(msg.data, BeaconBlock);
+      this.emit(BLOCK_TOPIC, block);
+    } catch (e) {
+    }
+  }
+  private handleIncomingAttestation(msg: any): void {
+    try {
+      const attestation: Attestation = deserialize(msg.data, Attestation);
+      this.emit(ATTESTATION_TOPIC, attestation);
+    } catch (e) {
+    }
+  }
+  private handleIncomingShardAttestation(msg: any): void {
+    try {
+      const attestation: Attestation = deserialize(msg.data, Attestation);
+      this.emit(shardAttestationTopic(attestation.data.shard), attestation);
+    } catch (e) {
+    }
+  }
+  private emitGossipHeartbeat(): void {
+    // this.emit("gossipsub:heartbeat");
+  }
+
+  // rpc
+  public getPeers(): PeerInfo[] {
+    return this.rpc.getPeers().map(hobbitaUri => {
+      return HobbitsUri.hobbitsUriToPeerInfo(hobbitaUri);
+    });
+  }
+
+  public hasPeer(peerInfo: PeerInfo): boolean {
+    return this.rpc.hasPeer(HobbitsUri.peerInfoToHobbitsUri(peerInfo));
+  }
+  public async connect(peerInfo: PeerInfo): Promise<void> {
+    await this.rpc.dialForRpc(HobbitsUri.peerInfoToHobbitsUri(peerInfo));
+  }
+  public async disconnect(peerInfo: PeerInfo): Promise<void> {
+    await this.rpc.removePeer(HobbitsUri.peerInfoToHobbitsUri(peerInfo));
+  }
+  // public async sendRequest<T extends ResponseBody>(peerInfo: PeerInfo, method: Method, body: RequestBody): Promise<T> {
+  //   return await this.rpc.sendRequest<T>(peerInfo, method, body);
```

Comment:
@wemeetagain I was asking for changing the type of `ResponseBody` and `RequestBody` here in order to support both `libp2p` and `hobbits`.